pyoozie/tags.py

```diff
@@ -624,6 +624,51 @@ def _xml(self, doc, tag, text, on_next, on_error):
         return doc
 
 
+class Decision(_WorkflowEntity):
+    """Node specifying a switch/case."""
+
+    def __init__(
+            self,
+            default,        # type: _WorkflowEntity
+            choices,        # type: typing.Dict[typing.Text, _WorkflowEntity]
+            name=None,      # type: typing.Optional[typing.Text]
+            on_error=None,  # type: typing.Optional[_WorkflowEntity]
+    ):
+        # type: (...) -> None
+        super(Decision, self).__init__(xml_tag='decision', name=name, on_error=on_error)
+        assert default, 'A default must be supplied'
+        assert choices, 'At least one choice required'
+        self.__default = default
+        self.__choices = choices
+
+    def _xml(self, doc, tag, text, on_next, on_error):
+        # type: (yattag.doc.Doc, yattag.doc.Doc.tag, yattag.doc.Doc.text, typing.Text, typing.Text) -> yattag.doc.Doc
+        _on_error = self._xml_and_get_on_error(doc, tag, text, on_next, on_error)
+
+        # Write switch/case
+        with tag(self.xml_tag, name=self.identifier()):
+            with tag('switch'):
+                for case, dest in self.__choices.items():
+                    with tag('case', to=dest.identifier()):
+                        doc.text(case)
+                doc.stag('default', to=self.__default.identifier())
+
+        # Write contained actions
+        self.__default._xml(doc, tag, text, on_next, _on_error)
+        for choice in self.__choices.values():
+            choice._xml(doc, tag, text, on_next, _on_error)
```

Comment:
That's fine, but my point was that it would be valid to, for example, have a transition to the error node on both error and a particular switch case.